circuits/kimchi/src/gates/endosclmul.rs

```diff
@@ -53,78 +55,48 @@ impl<F: FftField> CircuitGate<F> {
     ) -> Result<(), String> {
         let this: [F; COLUMNS] = array_init(|i| witness[i][self.row]);
         let next: [F; COLUMNS] = array_init(|i| witness[i][self.row + 1]);
-        let xq1 = (F::one() + ((cs.endo - F::one()) * next[12])) * this[0];
-        let xq2 = (F::one() + ((cs.endo - F::one()) * next[14])) * this[0];
 
-        ensure_eq!(self.typ, GateType::Endomul, "endomul: incorrect gate");
```

Comment:
I reverted this, since we check this for every gate. Not sure if it's useful but if you want to remove it remove all of them :)